### PR TITLE
compression: Add some doc comments

### PR DESCRIPTION
**Pull request by cgwalters**
I'm digging in more to zstd:chunked and I hope these comments are useful.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@giuseppe PTAL
Lint is not happy.

**Review by giuseppe — APPROVED** [review]
LGTM

File: pkg/chunked/internal/compression.go [hunk 1]
```diff
@@ -83,11 +107,22 @@ func GetType(t byte) (string, error) {
...
+	// <offset>:<length>:<uncompressed length>:<type>
+	// The <type> is ManifestTypeCRFS which should have the value `1`.
+	ManifestInfoKey = "io.github.containers.zstd-chunked.manifest-position"
+	// TarSplitInfoKey is an annotation that signals the start of the "tar-split" metadata
+	// contents which are embedded as a skippable zstd frame.  It has a format of
+	// three decimal integers separated by `:` as follows:
+	// <offset>:<length>:<uncompressed length>
+	TarSplitInfoKey = "io.github.containers.zstd-chunked.tarsplit-position"
+
+	// TarSplitChecksumKey is deprecated: Use the TOC.TarSplitDigest field instead, this annotation is no longer read nor written.
```
Comment (mtrmac) [inline pkg/chunked/internal/compression.go:124]: FWIW the exact `// Deprecated:` text is treated as special by Go tooling and linters.

(I know, that’s horrible.)

Comment (cgwalters) [reply pkg/chunked/internal/compression.go:124]: https://github.com/containers/storage/pull/1958

**Pull request merged**